Feature question es6

krispinn opened the pull request:
@steveortiz updated code to work on my machine. fixes are in latest commit : 772d3cf90771493b51436e8a12f9b8050a167b69

krispinn committed:
WIP Added intents

krispinn committed:
WIP added LIST_OF_ANSWERS_ROUND_ONE

krispinn committed:
WIP: got a simple question answer response working

krispinn committed:
WIP attempted refactor of original simple skill

krispinn committed:
WIP:
feature-question
Successfully refactored family fued to ask one question and verify results

krispinn committed:
fixing merge conflicts in package.json (??)

krispinn committed:
Removed javadocs haha

chesshacker reviewed (changes requested):
Remove .DS_Store files. You may want to create a global gitignore file. See [github docs on gitignore](https://help.github.com/articles/ignoring-files/)

chesshacker commented on src/action-intent-request.js:
```diff
@@ -0,0 +1,67 @@
+let model = require('./response-model'),
+    utilities = require('./utilities'),
+    ActionLaunchRequest = require('./action-launch-request'),
+    client = require('./api-client');
```
You can use `import model from './response-model';` etc.

krispinn replied:
@steveortiz  fixed here fb8cff278e58cce257bac97e9dbf3eb452d1ce71.
Not sure if you'd like me to do the same for the rest of the variables, utilities and so forth, but they aren't explicitly exported from the files with those names so I think I'd have to do a bit more refactoring for that

chesshacker commented on src/action-intent-request.js:
```diff
@@ -0,0 +1,67 @@
+  console.log('Intent Requested');
+
+  // get intent info
+  let intent = event.request.intent,
```
use `const` instead of `let` unless the variable is going to be changed to something else.

krispinn replied:
@steveortiz fixed here 09e6a5dacc1d4b5ab0a21f2f587b53402912a7b6

chesshacker commented on src/questionStorage.js:
```diff
@@ -0,0 +1,9 @@
+import roundQuestions from "./roundQuestions";
```
Rather than mixing camelCase and snake-case for the filenames, we should probably pick one. I like snake-case for the filenames as you did in most other cases. Will you rename this file `question-storage.js`?

krispinn replied:
@steveortiz renamed here 14d299cda35d9b77902d432d7979148b0574cebf and here 28a0959a30e08989003f254b38d3cad129351658

krispinn committed:
Removed .DS_Store

Added Console logs to api-client.js

chesshacker commented on src/index.js:
```diff
@@ -1,79 +1,33 @@
+  ActionIntentRequest = require('./action-intent-request'),
+  ActionSessionEndRequest = require('./action-session-end-request');
+
+exports.handler =  function (event, context){
```
FYI, this can be written in es6 as `export function(event, context) {`

krispinn replied:
@steveortiz  I originally tried to run the code in that way, however I was running into some issues when running npm-start. could you possibly help me with it tomorrow?

chesshacker commented on src/response-model.js:
```diff
@@ -0,0 +1,61 @@
+      "currentRound": 0
+    }
+
+    return sessionAttributes;
```
when you return something from a constructor, you override the class constructor... this probably isn't what you want to do with your class.

krispinn replied:
@steveortiz alright. Before without the return, I was getting an error that sessionAttributes didn't exists. Maybe we can look at it tomorrow, and for tonight I'll do some research on what else I can do.

chesshacker reviewed (commented):
Great job! I really like the way you reorganized things. I made a few comments, mostly to share a little ES6 style things.

krismndz commented:
@steveortiz .DS_Store removed here 48f194b4a1bf688a4f27960817ed56bcb8b4763e
working on configuring a .gitignore_global to avoid this issue in the furure

krispinn committed:
REFACTORE: feature-question-es6
Renamed questionStorage.js to question-storage.js

krispinn committed:
REFACTOR: feature-question-es6 filenames
Renamed roundQuestions.json to round-questions.json

krispinn committed:
REFACTOR: action-intent-request
removed let model = require('./response-model') and added import {model} from './reponse-model

krismndz commented:
@steveortiz  sounds good. working on the fixes now

krispinn committed:
REFACTOR: action-intent-request.js variable types
Changed 'let intent = ...' to 'const intent = ...'

krispinn committed:
REFACTOR:response-model
Changed export name to module.exports.ResponseModel = ResponseModel
REFACTOR: import files
Changes import file names to match the new snake case of the files

krispinn committed:
Removed commented out code

steveortiz commented:
This is looking better. Let's merge this and I'll add my unit tests to it.
